packages/c-pin-input/src/c-pin-input.tsx

```diff
@@ -0,0 +1,139 @@
+/**
+ * Hey! Welcome to @chakra-ui/vue-next CPinInput
+ *
+ * The pin input component is similar to the input component but is optimized for entering sequences of digits quickly
+ *
+ * @see Docs     https://next.vue.chakra-ui.com/c-pin-input
+ * @see Source   https://github.com/chakra-ui/chakra-ui-vue-next/blob/master/packages/c-pin-input/src/c-pin-input/c-pin-input.tsx
+ * @see WAI-ARIA https://www.w3.org/TR/wai-aria-practices-1.2
+ */
+
+import { h, defineComponent, computed, PropType, ComputedRef } from "vue"
+import {
+  chakra,
+  createStylesContext,
+  useStyleConfig,
+} from "@chakra-ui/vue-system"
+import {
+  createContext,
+  getValidChildren,
+  vueThemingProps,
+} from "@chakra-ui/vue-utils"
+import { CInput } from "@chakra-ui/c-input"
+import { connect } from "@zag-js/pin-input"
+import { PinInputProps, usePinInputMachine } from "./use-pin-input"
+
+const [StylesProvider, useStyles] = createStylesContext("CPinInput")
+const [PinInputProvider, usePinInput] =
+  createContext<ComputedRef<ReturnType<typeof connect>>>()
+
+export { PinInputProvider, usePinInput }
+
+export const CPinInputProps = {
+  value: {
+    type: Object as PropType<Array<string>>,
+    default: [],
+  },
+  id: {
+    type: String,
+    default: "0",
+  },
+  placeholder: {
+    type: String,
+    default: "o",
+  },
+  type: {
+    type: String as PropType<"alphanumeric" | "numeric">,
+    default: "numeric",
+  },
+  otp: {
+    type: Boolean,
+    default: false,
+  },
+  mask: {
+    type: Boolean,
+    default: false,
+  },
+  blurOnComplete: {
+    type: Boolean,
+    default: false,
+  },
+  dir: {
+    type: String as PropType<"rtl" | "ltr">,
+    default: "ltr",
+  },
+  spacing: {
+    type: [String, Number],
+    default: "0.75",
+  },
+  ...vueThemingProps,
+}
+
+export const CPinInput = defineComponent({
+  name: "CPinInput",
+  props: CPinInputProps,
+  emits: ["change", "invalid", "complete"],
```

Comment:
@Shyrro I think we can add `v-model` support by including the `modelValue` prop and emitting the `update:modelValue` event.

That way we could listen to the value of the `modelValue` prop and then update the pin input machine's value using the [`api.value.setValue(props.modelValue)`](https://zagjs.com/components/react/pin-input#methods-and-properties) method.

This would allow the final consumer to listen to the `v-model` event if they want to do real-time validation with their forms.

An example of this is in the `CCheckboxGroup` component. In this example I used a writable `computed` variable to achieve this effect.

**`v-model` event handling**
https://github.com/chakra-ui/chakra-ui-vue-next/blob/50cab7ddf68db6db45e7d6c607a74597dd1639cf/packages/c-checkbox/src/checkbox-group.tsx#L51-L58

**Emitted value processing** (In this case it should be simple for the pin-input. No fancy processing is required here)
https://github.com/chakra-ui/chakra-ui-vue-next/blob/50cab7ddf68db6db45e7d6c607a74597dd1639cf/packages/c-checkbox/src/use-checkbox-group.ts#L3-L19

**Final consumption**
https://github.com/chakra-ui/chakra-ui-vue-next/blob/50cab7ddf68db6db45e7d6c607a74597dd1639cf/packages/c-checkbox/examples/with-checkbox-group.vue#L1-L19